kustomize/commands/edit/add/addgenerator.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package add
+
+import (
+	"errors"
+	"log"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/kustomize/v4/commands/internal/kustfile"
+	"sigs.k8s.io/kustomize/kustomize/v4/commands/internal/util"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+type addGeneratorOptions struct {
+	generatorFilePaths []string
+}
+
+// newCmdAddGenerator adds the name of a file containing a generator
+// configuration to the kustomization file.
+func newCmdAddGenerator(fSys filesys.FileSystem) *cobra.Command {
+	var o addGeneratorOptions
+
+	cmd := &cobra.Command{
+		Use:   "generator",
+		Short: "Add the name of a file containing a generator configuration to the kustomization file.",
+		Example: `
+		add generator {filepath}`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate(args)
+			if err != nil {
+				return err
+			}
+			return o.RunAddGenerator(fSys)
+		},
+	}
+	return cmd
+}
+
+// Validate validates add generator command.
+func (o *addGeneratorOptions) Validate(args []string) error {
+	if len(args) == 0 {
+		return errors.New("must specify a generator file")
```

Comment:
nit: `must specify a yaml file which contains a generator plugin resource` or something like that. 

Unfortunately, we seem not have a doc to specify what `generator file` means. The ./example has generatorOption, configGenerations and some other "generate" examples that are not for `generator file`.

Reply:
done, both here and in `addtransformer.go`